internal/kafka/handler.go

```diff
@@ -41,8 +42,10 @@ func (this *handler) onMessage(ctx context.Context, msg *kafka.Message, cfg *con
 	}
 
 	// Validate RequestID
+	valuePayloadStatus := reflect.ValueOf(payloadStatus)
+	requestField := valuePayloadStatus.Elem().FieldByName("RequestID")
 	if cfg.RequestConfig.ValidateRequestIDLength != 0 {
-		if len(payloadStatus.RequestID) != cfg.RequestConfig.ValidateRequestIDLength {
+		if len(payloadStatus.RequestID) != cfg.RequestConfig.ValidateRequestIDLength || !requestField.IsValid() {
 			endpoints.IncInvalidConsumerRequestIDs()
 			return
 		}
```

Comment:
@aleccohan It maybe better to make `RequestID` of `message.PayloadStatusMessage` struct a string pointer (`*string`) rather than using reflect here. Then we can directly check if `payloadStatus.RequestID` is `null` rather than using the reflect library here. What do you think?

Reply:
Makes sense to me!  Changing now!

Reply:
I think we can handle this case even without making the `RequestID` a pointer.  I believe if the request_id isn't in the message, then it will get created as an empty string.  So the existing `len()` checks should catch the empty and null request_ids for us.  

I like to avoid using pointers whenever possible. (Pointers are cool, don't get me wrong!  :grin: )  However, with pointers, you'll need to check if the pointer is not nil before dereferencing otherwise you can run into issues where you dereference a nil pointer and bad things happen.  :)

Sometimes using pointers is really necessary (optional things, etc) and using a pointer for things like is a good idea!   So  :+1:  for looking ways to simplify things (reflect might be a bit "heavy-handed" here).

Reply:
Yep, makes sense! Less code for us to deal with this way ^ :100: